src/shared/molecules/MyDataTable/MyDataTable.tsx

```diff
@@ -299,16 +281,27 @@ const MyDataTable: React.FC<TProps> = ({
     showQuickJumper: true,
     showSizeChanger: true,
   };
-  const onSelectRowChange: SelectionSelectFn<TMyDataTableRow> = (
+  const onSelectRowChange: SelectionSelectFn<TMyDataTableRow> = async (
     record,
     selected
   ) => {
     const recordKey = record._self;
-
     const dataPanelLS: TResourceTableData = JSON.parse(
       localStorage.getItem(DATA_PANEL_STORAGE)!
     );
-    const localStorageRows = toLocalStorageResources(record, 'my-data');
+
+    let localStorageRows;
+    if (selected && isNil(record.distribution)) {
+      // Sometimes the distributions are not available in the response of the bulk fetch of resources. Delta is working on fixing this.
+      // In the mean time, fusion can retrieve this extra information by doing a separate request per resource that does not have `distribution`.
+      setFetchingResources(true);
+      const expandedResource = await fetchResourceForDownload(recordKey, nexus);
+      setFetchingResources(false);
+      localStorageRows = toLocalStorageResources(expandedResource, 'my-data');
+    } else {
+      localStorageRows = toLocalStorageResources(record, 'my-data');
+    }
+    toLocalStorageResources(record, 'my-data');
```

Comment:
I think we should allow the user to select whenever he want and deselect (showing loading state in a single resource selection/deselection may have a bad UX)
if a single request will take too much time (not likely will happen but we should consider it) then we can take care of canceling the request (we can discuss this)

Reply:
As discussed, the cancellation of request when fetching multiple resources will not be implemented atm because:

1. The whole table is "deactivated" with the loader when the user selects all, therefore, they don't have a change to de-select all the rows.
2. Delta will (hopefully) soon fix the issue of some resources not containing distributions field on their side. With that, we will not need this logic at all.